### PR TITLE
chore: more TS strict fixes

### PR DESCRIPTION
**Pull request by dbanksdesign**
#### Description of changes

* Some minor things to destructuring props
* Making TS happy in the splitPrimitiveProps function

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are updated
- [x] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated
- [x] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/react/src/primitives/Menu/MenuButton.tsx [hunk 1]
```diff
@@ -23,7 +23,7 @@ export const MenuButton = React.forwardRef<
       isDisabled,
       isLoading,
       size,
-      style,
+      style = {},
```
Comment (calebpollman) [inline packages/react/src/primitives/Menu/MenuButton.tsx:26]: As `style` is used in the dep array for the memoized return value of `useStyles` (see [source](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/src/primitives/shared/styleUtils.ts#L164)), we should avoid setting a default parameter of an empty object here. The empty object will have a new reference on every render, which causes a re-evaluation of the aforementioned `useMemo` in `useStyles`

**Comment by reesscot** [comment]
Related to https://github.com/aws-amplify/amplify-ui/issues/1018

File: packages/react/src/primitives/shared/styleUtils.ts [hunk 2]
```diff
@@ -167,31 +167,47 @@ export const useStyles = (
...
-  rest: Omit<
-    PrimitiveProps,
-    keyof FlexContainerStyleProps | keyof BaseStyleProps
+  rest: Partial<
+    Omit<PrimitiveProps, keyof FlexContainerStyleProps | keyof BaseStyleProps>
   >;
 }
 
+function isStyleKey(prop: any): prop is keyof AllStyleProps {
+  if (prop in ComponentPropsToStylePropsMap) {
```
Comment (reesscot) [inline packages/react/src/primitives/shared/styleUtils.ts:176]: Would just returning `prop in ComponentPropsToStylePropsMap` also work? Seems like that should already return a boolean.

File: packages/react/src/primitives/types/tabs.ts [hunk 3]
```diff
@@ -60,4 +60,10 @@ export interface TabItemProps extends BaseComponentProps, BaseStyleProps {
    * Make a Tab not clickable and its content not visible to the user. Defaults to false (i.e., visible).
    */
   isDisabled?: boolean;
+
+  /**
+   * @description
+   * Accepts any number of TabItem components
+   */
+  children?: React.ReactNode;
```
Comment (reesscot) [inline packages/react/src/primitives/types/tabs.ts:68]: Does this fix this issue? https://github.com/aws-amplify/amplify-ui/issues/3256

**Review by wlee221 — APPROVED** [review]

**Review by ErikCH — APPROVED** [review]

**Pull request merged**